contracts/meta-oracles/feeds/TwoAssetLinearizedTimeSeriesFeed.sol

```diff
@@ -0,0 +1,199 @@
+
+    // Amount of time after which read interpolates price result, in seconds
+    uint256 public interpolationThreshold; 
+    string public dataDescription;
+    IOracle public baseOracleInstance;
+    IOracle public quoteOracleInstance;
+
+
+    /* ============ Events ============ */
+
+    event LogOracleUpdated(
+        address indexed newOracleAddress
+    );
+
+    /* ============ Constructor ============ */
+
+    /*
+     * Set interpolationThreshold, data description, quote oracle and base oracle and instantiate oracle
+     *
+     * @param  _updateInterval            Cadence at which data is optimally logged. Optimal schedule is based
+                                          off deployment timestamp. A certain data point can't be logged before
+                                          it's expected timestamp but can be logged after (for TimeSeriesFeed)
+     * @param  _nextEarliestUpdate        Time the first on-chain price update becomes available (for TimeSeriesFeed)
+     * @param  _maxDataPoints             The maximum amount of data points the linkedList will hold (for TimeSeriesFeed)
+     * @param  _seededValues              Array of previous timeseries values to seed initial values in list.
+     *                                    The last value should contain the most current piece of data (for TimeSeriesFeed)
+     * @param  _interpolationThreshold    The minimum time in seconds where interpolation is enabled
+     * @param  _baseOracleAddress         The address of the base oracle to read current data from
+     * @param  _quoteOracleAddress        The address of the quote oracle to read current data from
+     * @param  _dataDescription           Description of contract for Etherscan / other applications
+     */
+    constructor(
+        uint256 _updateInterval,
+        uint256 _nextEarliestUpdate,
+        uint256 _maxDataPoints,
+        uint256[] memory _seededValues,
+        uint256 _interpolationThreshold,
+        IOracle _baseOracleAddress,
+        IOracle _quoteOracleAddress,
+        string memory _dataDescription
+    )
+        public
+        TimeSeriesFeedV2(
+            _updateInterval,
+            _nextEarliestUpdate,
+            _maxDataPoints,
+            _seededValues
+        )
+    {
+        interpolationThreshold = _interpolationThreshold;
+        baseOracleInstance = _baseOracleAddress;
+        quoteOracleInstance = _quoteOracleAddress;
+        dataDescription = _dataDescription;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Change base asset oracle in case current one fails or is deprecated. Only contract
+     * owner is allowed to change.
+     *
+     * @param  _newBaseOracleAddress       Address of new oracle to pull data from
+     */
+    function changeBaseOracle(
+        IOracle _newBaseOracleAddress
+    )
+        external
+        onlyOwner
+        timeLockUpgrade // Must be placed after onlyOwner
```

Comment:
Let's use `timeLockUpgradeV2` from now on. I'm exporting it from the latest version of `set-protocol-contracts:1.2.9`.

It'll automatically be only callable by the owner